### ISSUE TITLE
model validator ignores message code

### ISSUE THREAD

**Issue by rtiagoViz** [phalcon/cphalcon#12645]
Model validation
```php
    public function validation()
    {

        $validator = new Validation();

        $validator->add(
            'email',
            new Uniqueness([
                'model' => $this,
                'message' => "The email must be unique.",
                'code' => 20
            ])
        );

        $this->validate($validator);
```

Uniquess validator properly sets the code but model.zep below ignores it.

```zep
	protected function validate(<ValidationInterface> validator) -> boolean
	{
		var messages, message;

		let messages = validator->validate(null, this);

		// Call the validation, if it returns not the boolean
		// we append the messages to the current object
		if typeof messages == "boolean" {
			return messages;
		}

		for message in iterator(messages) {
			this->appendMessage(
				new Message(
					message->getMessage(),
					message->getField(),
					message->getType()
				)
			);
		}

		// If there is a message, it returns false otherwise true
		return !count(messages);
	}
```

**Comment by dreamsxin** [comment]
You can help fix it, Pull request.

**Comment by sergeyklay** [comment]
Cc: @Jurigag

**Comment by Jurigag** [comment]
Tbh it's nothing that i was changing, that someone in model validate method forgot to add 4th argument in Message constructor.

**Comment by Jurigag** [comment]
Fixed in 3.1.x
